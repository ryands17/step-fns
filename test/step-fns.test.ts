// import { SynthUtils } from '@aws-cdk/assert'
// import * as cdk from '@aws-cdk/core'
// import * as StepFns from '../lib/step-fns-stack'

test('Snapshot matches resources', () => {
  // const stack = createStack({ outdir: 'cdk.out', stackTraces: false })
  // expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot()

  // TODO: replace snapshot testing
  expect(true).toBeTruthy()
})

// function createStack(props?: cdk.AppProps) {
//   const app = new cdk.App(props)
//   return new StepFns.StepFnsStack(app, 'MyTestStack')
// }
