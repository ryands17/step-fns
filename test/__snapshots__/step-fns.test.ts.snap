// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot matches resources 1`] = `
Object {
  "Parameters": Object {
    "AssetParameters67b7823b74bc135986aa72f889d6a8da058d0c4a20cbc2dfc6f78995fdd2fc24ArtifactHashBA91B77F": Object {
      "Description": "Artifact hash for asset \\"67b7823b74bc135986aa72f889d6a8da058d0c4a20cbc2dfc6f78995fdd2fc24\\"",
      "Type": "String",
    },
    "AssetParameters67b7823b74bc135986aa72f889d6a8da058d0c4a20cbc2dfc6f78995fdd2fc24S3Bucket4D46ABB5": Object {
      "Description": "S3 bucket for asset \\"67b7823b74bc135986aa72f889d6a8da058d0c4a20cbc2dfc6f78995fdd2fc24\\"",
      "Type": "String",
    },
    "AssetParameters67b7823b74bc135986aa72f889d6a8da058d0c4a20cbc2dfc6f78995fdd2fc24S3VersionKeyB0F28861": Object {
      "Description": "S3 key for asset version \\"67b7823b74bc135986aa72f889d6a8da058d0c4a20cbc2dfc6f78995fdd2fc24\\"",
      "Type": "String",
    },
    "AssetParametersc9f2da6914ad1d0e0c31426b3fa0f686d4d30b96f2101ab412a889aa712324caArtifactHash90E2D53E": Object {
      "Description": "Artifact hash for asset \\"c9f2da6914ad1d0e0c31426b3fa0f686d4d30b96f2101ab412a889aa712324ca\\"",
      "Type": "String",
    },
    "AssetParametersc9f2da6914ad1d0e0c31426b3fa0f686d4d30b96f2101ab412a889aa712324caS3Bucket875981DA": Object {
      "Description": "S3 bucket for asset \\"c9f2da6914ad1d0e0c31426b3fa0f686d4d30b96f2101ab412a889aa712324ca\\"",
      "Type": "String",
    },
    "AssetParametersc9f2da6914ad1d0e0c31426b3fa0f686d4d30b96f2101ab412a889aa712324caS3VersionKeyF8E4C299": Object {
      "Description": "S3 key for asset version \\"c9f2da6914ad1d0e0c31426b3fa0f686d4d30b96f2101ab412a889aa712324ca\\"",
      "Type": "String",
    },
    "AssetParameterse14f1b160ac97eea9a52231485b365ea3f1704568c0e53911fc6e93870d90c29ArtifactHash89E94185": Object {
      "Description": "Artifact hash for asset \\"e14f1b160ac97eea9a52231485b365ea3f1704568c0e53911fc6e93870d90c29\\"",
      "Type": "String",
    },
    "AssetParameterse14f1b160ac97eea9a52231485b365ea3f1704568c0e53911fc6e93870d90c29S3Bucket55097FDE": Object {
      "Description": "S3 bucket for asset \\"e14f1b160ac97eea9a52231485b365ea3f1704568c0e53911fc6e93870d90c29\\"",
      "Type": "String",
    },
    "AssetParameterse14f1b160ac97eea9a52231485b365ea3f1704568c0e53911fc6e93870d90c29S3VersionKey6D2268C1": Object {
      "Description": "S3 key for asset version \\"e14f1b160ac97eea9a52231485b365ea3f1704568c0e53911fc6e93870d90c29\\"",
      "Type": "String",
    },
    "AssetParametersff43ba4241ebd6f44892a46e3da58f261a8a3a4cd4c826a054cdb54b20cbf34fArtifactHash06FE406E": Object {
      "Description": "Artifact hash for asset \\"ff43ba4241ebd6f44892a46e3da58f261a8a3a4cd4c826a054cdb54b20cbf34f\\"",
      "Type": "String",
    },
    "AssetParametersff43ba4241ebd6f44892a46e3da58f261a8a3a4cd4c826a054cdb54b20cbf34fS3Bucket88A980CC": Object {
      "Description": "S3 bucket for asset \\"ff43ba4241ebd6f44892a46e3da58f261a8a3a4cd4c826a054cdb54b20cbf34f\\"",
      "Type": "String",
    },
    "AssetParametersff43ba4241ebd6f44892a46e3da58f261a8a3a4cd4c826a054cdb54b20cbf34fS3VersionKey5387DB30": Object {
      "Description": "S3 key for asset version \\"ff43ba4241ebd6f44892a46e3da58f261a8a3a4cd4c826a054cdb54b20cbf34f\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A": Object {
      "DependsOn": Array [
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters67b7823b74bc135986aa72f889d6a8da058d0c4a20cbc2dfc6f78995fdd2fc24S3Bucket4D46ABB5",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters67b7823b74bc135986aa72f889d6a8da058d0c4a20cbc2dfc6f78995fdd2fc24S3VersionKeyB0F28861",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters67b7823b74bc135986aa72f889d6a8da058d0c4a20cbc2dfc6f78995fdd2fc24S3VersionKeyB0F28861",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:PutRetentionPolicy",
                "logs:DeleteRetentionPolicy",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "Roles": Array [
          Object {
            "Ref": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "addDEC2DAFB": Object {
      "DependsOn": Array [
        "addServiceRole725BBB92",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersff43ba4241ebd6f44892a46e3da58f261a8a3a4cd4c826a054cdb54b20cbf34fS3Bucket88A980CC",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersff43ba4241ebd6f44892a46e3da58f261a8a3a4cd4c826a054cdb54b20cbf34fS3VersionKey5387DB30",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersff43ba4241ebd6f44892a46e3da58f261a8a3a4cd4c826a054cdb54b20cbf34fS3VersionKey5387DB30",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "NODE_OPTIONS": "--enable-source-maps",
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "addServiceRole725BBB92",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "addLogRetentionFF2DF5AC": Object {
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "addDEC2DAFB",
              },
            ],
          ],
        },
        "RetentionInDays": 7,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "addServiceRole725BBB92": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "chainingA7D7050E": Object {
      "DependsOn": Array [
        "chainingRoleDefaultPolicy910C3AB8",
        "chainingRole7D073841",
      ],
      "Properties": Object {
        "DefinitionString": Object {
          "Fn::Join": Array [
            "",
            Array [
              "{\\"StartAt\\":\\"add 2 numbers\\",\\"States\\":{\\"add 2 numbers\\":{\\"Next\\":\\"isBigNumber\\",\\"Retry\\":[{\\"ErrorEquals\\":[\\"Lambda.ServiceException\\",\\"Lambda.AWSLambdaException\\",\\"Lambda.SdkClientException\\"],\\"IntervalSeconds\\":2,\\"MaxAttempts\\":6,\\"BackoffRate\\":2}],\\"Type\\":\\"Task\\",\\"Resource\\":\\"",
              Object {
                "Fn::GetAtt": Array [
                  "addDEC2DAFB",
                  "Arn",
                ],
              },
              "\\"},\\"isBigNumber\\":{\\"Type\\":\\"Choice\\",\\"Choices\\":[{\\"Variable\\":\\"$\\",\\"NumericGreaterThan\\":50,\\"Next\\":\\"double the bigger number\\"}],\\"Default\\":\\"double the number\\"},\\"double the number\\":{\\"End\\":true,\\"Retry\\":[{\\"ErrorEquals\\":[\\"Lambda.ServiceException\\",\\"Lambda.AWSLambdaException\\",\\"Lambda.SdkClientException\\"],\\"IntervalSeconds\\":2,\\"MaxAttempts\\":6,\\"BackoffRate\\":2},{\\"ErrorEquals\\":[\\"NumberIsTooBig\\"],\\"MaxAttempts\\":0}],\\"Catch\\":[{\\"ErrorEquals\\":[\\"NumberIsTooBig\\"],\\"Next\\":\\"double the bigger number\\"}],\\"Type\\":\\"Task\\",\\"Resource\\":\\"",
              Object {
                "Fn::GetAtt": Array [
                  "double71C0A51C",
                  "Arn",
                ],
              },
              "\\"},\\"double the bigger number\\":{\\"End\\":true,\\"Retry\\":[{\\"ErrorEquals\\":[\\"Lambda.ServiceException\\",\\"Lambda.AWSLambdaException\\",\\"Lambda.SdkClientException\\"],\\"IntervalSeconds\\":2,\\"MaxAttempts\\":6,\\"BackoffRate\\":2}],\\"Type\\":\\"Task\\",\\"Resource\\":\\"",
              Object {
                "Fn::GetAtt": Array [
                  "doubleBigNumber2ACC4FF4",
                  "Arn",
                ],
              },
              "\\"}},\\"TimeoutSeconds\\":120}",
            ],
          ],
        },
        "RoleArn": Object {
          "Fn::GetAtt": Array [
            "chainingRole7D073841",
            "Arn",
          ],
        },
      },
      "Type": "AWS::StepFunctions::StateMachine",
    },
    "chainingRole7D073841": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "states.",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ".amazonaws.com",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "chainingRoleDefaultPolicy910C3AB8": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "addDEC2DAFB",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "double71C0A51C",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "doubleBigNumber2ACC4FF4",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "chainingRoleDefaultPolicy910C3AB8",
        "Roles": Array [
          Object {
            "Ref": "chainingRole7D073841",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "double71C0A51C": Object {
      "DependsOn": Array [
        "doubleServiceRoleCE6D9CC3",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameterse14f1b160ac97eea9a52231485b365ea3f1704568c0e53911fc6e93870d90c29S3Bucket55097FDE",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameterse14f1b160ac97eea9a52231485b365ea3f1704568c0e53911fc6e93870d90c29S3VersionKey6D2268C1",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameterse14f1b160ac97eea9a52231485b365ea3f1704568c0e53911fc6e93870d90c29S3VersionKey6D2268C1",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "NODE_OPTIONS": "--enable-source-maps",
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "doubleServiceRoleCE6D9CC3",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "doubleBigNumber2ACC4FF4": Object {
      "DependsOn": Array [
        "doubleBigNumberServiceRole456F5997",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersc9f2da6914ad1d0e0c31426b3fa0f686d4d30b96f2101ab412a889aa712324caS3Bucket875981DA",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersc9f2da6914ad1d0e0c31426b3fa0f686d4d30b96f2101ab412a889aa712324caS3VersionKeyF8E4C299",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersc9f2da6914ad1d0e0c31426b3fa0f686d4d30b96f2101ab412a889aa712324caS3VersionKeyF8E4C299",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "NODE_OPTIONS": "--enable-source-maps",
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "doubleBigNumberServiceRole456F5997",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "doubleBigNumberLogRetention27D9FAF4": Object {
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "doubleBigNumber2ACC4FF4",
              },
            ],
          ],
        },
        "RetentionInDays": 7,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "doubleBigNumberServiceRole456F5997": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "doubleLogRetentionCD092049": Object {
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "double71C0A51C",
              },
            ],
          ],
        },
        "RetentionInDays": 7,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "doubleServiceRoleCE6D9CC3": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
